internal/factories/factories.go

```diff
@@ -56,6 +56,11 @@ type ReceiverFactory interface {
 		consumer consumer.MetricsConsumer) (receiver.MetricsReceiver, error)
 }
 
+// ErrDataTypeIsNotSupported can be returned by CreateTraceReceiver or
+// CreateMetricsReceiver if the particular data type is not supported by
+// the receiver.
+var ErrDataTypeIsNotSupported = errors.New("data type is not supported")
```

Comment:
Nit: I think "telemetry type" may be clearer here.

Reply:
Good point, I will change.